src/ucp/core/ucp_worker.c

```diff
@@ -1658,16 +1728,29 @@ ucs_status_t ucp_worker_create(ucp_context_h context,
         goto err_close_ifaces;
     }
 
+    /* Open all resources as connection managers on this worker */
+    status = ucp_worker_add_resource_cms(worker);
+    if ((status != UCS_OK) && (status != UCS_ERR_UNSUPPORTED)) {
+        goto err_close_cms;
+    }
+
+    if ((worker->num_cms == 0) &&
```

Comment:
if we got here, ucp_worker_add_resource_cms returned UCS_OK. why need to check num_cms again?